phys2bids/reporting/html_report.py

```diff
@@ -0,0 +1,219 @@
+
+
+def _generate_bokeh_plots(phys_in, size=(250, 500)):
+    """
+    Plot all the channels for visualizations as linked line plots for dynamic report.
+
+    Parameters
+    ----------
+    phys_in: BlueprintInput object
+        Object returned by BlueprintInput class
+    figsize: tuple
+        Size of the figure expressed as (size_x, size_y),
+        Default is 250x750px
+    -----
+    outcome:
+        Creates new plot with path specified in outfile.
+
+    See Also
+    --------
+    https://phys2bids.readthedocs.io/en/latest/howto.html
+    """
+    colors = ['#ff7a3c', '#008eba', '#ff96d3', '#3c376b', '#ffd439']
+
+    time = phys_in.timeseries.T[0]  # assumes first phys_in.timeseries is time
+    ch_num = len(phys_in.ch_name)
+    if ch_num > len(colors):
+        colors *= 2
+
+    downsample = int(phys_in.freq / 100)
+    plot_list = []
+    for row, timeser in enumerate(phys_in.timeseries.T[1:]):
+        # build a data source for each plot, with only the data + index (time)
+        # for the purpose of reporting, data is downsampled 10x
+        # doesn't make much of a difference to the naked eye, fine for reports
+        source = ColumnDataSource(data=dict(
+            x=time[::downsample],
+            y=timeser[::downsample]))
+
+        i = row + 1
+
+        tools = ['wheel_zoom,pan,reset']
+        q = figure(plot_height=size[0], plot_width=size[1],
+                   tools=tools,
+                   title=f' Channel {i}: {phys_in.ch_name[i]}',
+                   sizing_mode='stretch_both',
+                   x_range=(0, 100))
+        q.line('x', 'y', color=colors[i - 1], alpha=0.9, source=source)
+        q.xaxis.axis_label = 'Time (s)'
+        # hovertool commented for posterity because I (KB) will be triumphant
+        # eventually
+        # q.add_tools(HoverTool(tooltips=[
+        #    (phys_in.ch_name[i], '@y{0.000} ' + phys_in.units[i]),
+        #    ('HELP', '100 :D')
+        # ], mode='vline'))
+        plot_list.append([q])
+    p = gridplot(plot_list, toolbar_location='right',
+                 plot_height=250, plot_width=750,
+                 merge_tools=True)
+    script, div = components(p)
+    return script, div
+
+
+def generate_report(out_dir, log_path, phys_in):
+    """
+    Plot all the channels for visualizations as linked line plots for dynamic report.
+
+    Parameters
+    ----------
+    out_dir : str
+        File path to a completed phys2bids output directory
+    log_path: path
+        Path to the logged output of phys2bids
+    phys_in: BlueprintInput object
+        Object returned by BlueprintInput class
+    -----
+    outcome:
+        Creates new plot with path specified in outfile.
+
+    See Also
+    --------
+    https://phys2bids.readthedocs.io/en/latest/howto.html
+    """
+    # Copy assets into output folder
+    pkgdir = sys.modules['phys2bids'].__path__[0]
+    assets_path = opj(pkgdir, 'reporting', 'assets')
+    copy_tree(assets_path, opj(out_dir, 'assets'))
```

Comment:
Question: are we sure that a folder `assets` in the root dir of a BIDS dataset is BIDS compatible?
Maybe we should move it in `code/conversion` (or whatever we're using already), or add/append to `.gitignore` this folder.

Reply:
No idea... But it is a necessary folder. Moving to the `code/conversion` folder sounds good to me but I wonder what the best practices are in such situations.

Maybe @tsalo knows?

Reply:
I  Guys, Nice question, you should put every non standard thing in the ``code/conversion`` folder, it is actually thought for this kind of things such as logs and reports

Reply:
Done! Thank you guys!